src/kinetics/Reaction.cpp

```
@@ -488,365 +623,198 @@ bool Reaction::usesElectrochemistry(const Kinetics& kin) const
+        name = third_body.substr(2, third_body.size() - 3);
     }
-}
 
-double ThirdBody::efficiency(const std::string& k) const
-{
-    return getValue(efficiencies, k, default_efficiency);
+    if (name == m_name) {
+        return;
+    }
+    if (name == "M") {
+        throw CanteraError("ThirdBody::setName",
+            "Unable to revert explicit third body '{}' to 'M'", m_name);
+    }
+    if (efficiencies.size()) {
+        throw CanteraError("ThirdBody::setName",
+            "Conflicting efficiency definition for explicit third body '{}'", name);
+    }
+    m_name = name;
+    if (name == "<multiple>") {
+        // sentinel value for multiple third body colliders
+        return;
+    }
+    default_efficiency = 0.;
+    efficiencies[m_name] = 1.;
 }
 
-ThreeBodyReaction::ThreeBodyReaction()
+ThirdBody::ThirdBody(const AnyMap& node)
 {
-    m_third_body.reset(new ThirdBody);
-    setRate(newReactionRate(type()));
+    setParameters(node);
 }
 
-ThreeBodyReaction::ThreeBodyReaction(const Composition& reactants,
-                                     const Composition& products,
-                                     const ArrheniusRate& rate,
-                                     const ThirdBody& tbody)
-    : Reaction(reactants, products, make_shared<ArrheniusRate>(rate))
+void ThirdBody::setEfficiencies(const AnyMap& node)
 {
-    m_third_body = std::make_shared<ThirdBody>(tbody);
+    warn_deprecated("ThirdBody::setEfficiencies", node,
+        "To be removed after Cantera 3.0. Renamed to setParameters");
+    setParameters(node);
 }
 
-ThreeBodyReaction::ThreeBodyReaction(const AnyMap& node, const Kinetics& kin)
+void ThirdBody::setParameters(const AnyMap& node)
 {
-    m_third_body.reset(new ThirdBody);
-    if (!node.empty()) {
-        setParameters(node, kin);
-        setRate(newReactionRate(node, calculateRateCoeffUnits(kin)));
-    } else {
-        setRate(newReactionRate(type()));
+    if (m_name != "M") {
+        return;
+    }
```

Why does this silently return without changing anything?

---

This should raise an error ...

---

Removed